### PR TITLE
Auth reset pages UI

### PR DESCRIPTION
**Pull request by yashgajaria**
## Notion ticket link

[Reset Password](https://www.notion.so/uwblueprintexecs/Reset-Password-614cef5ae84846f7ac653bd6bf5a1398)

## Implementation description
* Reset password page has different stages that are the "inner" components of the page
* As you click through the screens, the different inner stages increment to change the inner part of the page
* The components (button-/components/button/index.tsx, checkbox- src/components/checkbox/index.tsx, textInput- src/components/textInput/index.tsx, and passwordInput- src/components/passwordInput/index.tsx) are part of another PR 

## Steps to test
1. Go to "/reset" and view page in mobile view. (Pages have been designed for mobile view so far) 
2. Can't move on from the "Reset Password" unless email is entered (note you need to click away from the input field
 after a valid email is entered)
image

3. Check Email page, the Open Email App button doesn't work yet (still investigating on if it's possible to directly open the email app from a webpage on mobile)
image

4. Password validation just checks if the two fields match and the password is more than 7 characters. 
5. Final screen has a button that redirects to login. 

## What should reviewers focus on?
* Only the /reset page 
* Just the overall look and feel of the screens in the reset password flow
* If anyone has input on how to directly open the mail app from the button on the 'Check Email' screen that'd be great  

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/reset.tsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                errorMessage={passwordInvalid.reason}
+                isRequired
+                onChange={(event) => {
+                    setConfirmPassword(event.currentTarget.value);
+                }}
+                onBlur={() => {
+                    if (confirmPassword != newPassword) {
+                        setPasswordInvalid({
+                            isInvalid: true,
+                            reason: "Passwords Do Not Match",
```
Comment (Aphexis) [inline pages/reset.tsx:137]: nit: should each word be capitalized?

File: pages/reset.tsx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            setPasswordRequirements({
+                isInvalid: true,
+                reason: "Password does not meet requirements",
+            });
+            setCanContinue(false);
+        }
+    };
+
+    const incrementStage = () => {
+        // add logic to check if it's possible
```
Comment (Aphexis) [inline pages/reset.tsx:56]: is this todo? otherwise can remove (lol this is also probably transferred from my old wip)

File: pages/reset.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                    tabindex="0"
+                    role="button"
+                    onClick={decrementStage}
+                    mb="-5"
+                    mt="2"
+                >
+                    <ChevronLeftIcon />
+                    Back
+                </Text>
+            )}
```
Comment (Aphexis) [inline pages/reset.tsx:230]: might be nice to have a bit more padding on the "< Back"

File: pages/reset.tsx [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+                            </Link>
+                        </Text>
+                    </Center>
+                    <Text mt="70%">
+                        Did not recieve the email? Check your spam folder or{" "}
+                        <Link onClick={decrementStage}>
+                            <Text as="u">try again.</Text>
+                            {/* Verify whether try again takes users back or will initiate "send email again" flow */}
+                        </Link>
+                    </Text>
```
Comment (Aphexis) [inline pages/reset.tsx:193]: should this be positioned at the lower/on the bottom of the screen?

**Review by Aphexis — COMMENTED** [review]
looks good, just minor comments on cleaning up the code diff + styling

File: pages/reset.tsx (hunk 2 above)
Comment (yashgajaria) [reply pages/reset.tsx:56]: I think we may want validation later about like has the page been completed before moving on- but I think we prevent that by disabling the button so we are fine?

File: pages/reset.tsx (hunk 4 above)
Comment (yashgajaria) [reply pages/reset.tsx:193]: Yup- this is positioned at the bottom of the screen

File: src/components/main/index.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import React, { FunctionComponent } from "react";
 import { Box, useTheme } from "@chakra-ui/react";
 
-import { Button } from "@components";
+import { AuthButton } from "@components";
```
[the remark this answers is not in the document]
Comment (yashgajaria) [reply src/components/main/index.tsx:4]: We only export AuthButton and not Button- do we want to export both?

[the remark this answers is not in the document]
Comment (yashgajaria) [reply src/components/main/index.tsx:4]: actually i dont think its possible- since we over-wrote the old button and renamed it authbutton

**Pull request closed without merging**